README.md

```diff
@@ -2,88 +2,139 @@
 
-#Push to dockerhub
-docker push <desired-name>
-```
+##API Design
+
+A swagger.yaml file is provided that documents the API per the OpenAPI specification.
+
+##Key Components
+
+####Environments
+
+An environment consists of a kubernetes namespace and our specific secrets associated with it. Each environment comes with a `routing` secret that contains two key-value pairs, a `public-api-key` and a `private-api-key`. These are for use with the [k8s-pods-ingress](https://github.com/30x/k8s-pods-ingress) to allow for secure communication with pods from inside and outside of the kubernetes cluster.  
+
+
+##Apigee Specific Annotations
+
+####Environments
+
+When created environments can accept an array of valid host names to accept traffic from. This array is represented on the namespace object as a space delimited annotation. The individual values must be either a valid IP address or valid host name. 
+
+####Deployments
+
+When created deployments can accept a `publicHosts` value, a `privateHosts` value or both. These values are for use with the [k8s-pods-ingress](https://github.com/30x/k8s-pods-ingress) and are the host name where the deployment can be reached. These values are stored as annotations on the deployed pods. 
+
+####Pod Template Specs
+
+When they are provided to the deployments endpoint pod template specs must have several Apigee specific labels and annotations.  
+
+**Labels:**
+For pods to be recognized by the [k8s-pods-ingress](https://github.com/30x/k8s-pods-ingress) they must have a label named `"routable"` with a value of `"true"`.
+
+**Annotations:**
+For pods to be properly routed by the [k8s-pods-ingress](https://github.com/30x/k8s-pods-ingress) they must have a `"publicPaths"` and/or `"privatePaths"` annotation where the value is of the form `{PORT}:{PATH}`. You may have multiple space delimited `{PORT}:{PATH}` combinations on each annotation. 
+ 
+
+##Usage
+
+> This assumes you are running the server locally, it is accessible at localhost:9000, and your kubernetes cluster is exposed with `kubectl proxy --port=8080`
+
+**Note:** All API calls require a valid JWT to be passed into an authorization header. For these examples we are using an empty JSON object that has been base64 encoded. 
 
-The server can now be deployed using:
+####Create a new environment:
 
 ```sh
-kubectl run enrober --image=<desired-name>
+curl -X POST -d '{
+	"environmentName": "org1-env1",
+	"hostNames": ["host1"]
+	}' \
+"localhost:9000/environments"
 ```
 
-##API Design
+This will create a `org1-env1` namespace and a secret named `routing` with two key-value pairs:
 
-A swagger.yaml file is provided that documents the API per the OpenAPI specification.
+- public-api-key
+- private-api-key
 
-###Usage
+The value of each of these keys-value pairs will a 256-bit base64 encoded randomized string. These secrets are for use with [30x/k8s-pods-ingress](https://github.com/30x/k8s-pods-ingress)
```

Comment:
All references to `k8s-pods-ingress` should be `k8s-router` and all references to `https://github.com/30x/k8s-pods-ingress` should be `https://github.com/30x/k8s-router`.  This happened a few times in this document, not just this one.